### PR TITLE
chore: refactor account ID

### PR DESCRIPTION
**Pull request by chris13524**
# Description

I took a look at the specs and noticed that we actually know the format of all CAIP-10 account IDs: `namespace:reference:address`. So made a master regex which extracts these three components. This now validates the length of the entire string, and the reference. Also generally refactors the account ID validation logic into some more files and refactor the tests some. Generally sets us up better for Solana support.

[Related Slack conversation](https://walletconnect.slack.com/archives/CB3MNK9C7/p1708368865218169)

## How Has This Been Tested?

Existing/modified tests

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

**Review by geekbrother — APPROVED** [review]
lgtm! Nice tests coverage!

File: src/model/types/account_id/caip10.rs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    #[error("Account ID uses an unsupported namespace")]
+    UnsupportedNamespace,
+
+    #[error("Account ID is eip155 namespace but: {0}")]
+    Eip155(#[from] Eip155Error),
+}
+
+// https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-10.md#syntax
+static PATTERN: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^([-a-z0-9]{3,8}):([-_a-zA-Z0-9]{1,32}):([-.%a-zA-Z0-9]{1,128})$").unwrap()
```
Comment (nopestack) [inline src/model/types/account_id/caip10.rs:22]: is it safe to unwrap here?

File: src/model/types/account_id/eip155.rs [hunk 2]
```diff
@@ -0,0 +1,155 @@
+use {
+    crate::model::types::erc55::erc_55_checksum_encode, once_cell::sync::Lazy, regex::Regex,
+    thiserror::Error,
+};
+
+// https://github.com/ChainAgnostic/namespaces/blob/main/eip155/caip10.md
+pub const NAMESPACE_EIP155: &str = "eip155";
+
+// https://github.com/ChainAgnostic/namespaces/blob/main/eip155/caip2.md#syntax
+static REFERENCE_PATTERN: Lazy<Regex> = Lazy::new(|| Regex::new(r"^\d+$").unwrap());
```
Comment (nopestack) [inline src/model/types/account_id/eip155.rs:10]: same question about unwraps here

File: src/model/types/account_id/eip155.rs [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        let reference = "abc";
+        assert_eq!(
+            validate_eip155(reference, address),
+            Err(Eip155Error::Reference(Eip155ReferenceError::Regex))
+        );
+    }
+
+    #[test]
+    fn test_is_valid_eip155_account() {
+        assert!(validate_caip_10("eip155:1:0x62639418051006514eD5Bb5B20aa7aAD642cC2d0").is_ok());
```
Comment (nopestack) [inline src/model/types/account_id/eip155.rs:114]: consider turning these strings into a constant

File: src/model/types/account_id/caip10.rs (hunk 1 above)
Comment (chris13524) [reply src/model/types/account_id/caip10.rs:22]: Hmm I suppose this only evaluates on first call, not startup. However failures in parsing the regex should all be caught by test cases. Will change to `.expect()` with that explanation

File: src/model/types/account_id/eip155.rs (hunk 3 above)
Comment (chris13524) [reply src/model/types/account_id/eip155.rs:114]: I don't think a constant would make these test cases easier to understand

File: src/model/types/account_id/eip155.rs (hunk 2 above)
Comment (chris13524) [reply src/model/types/account_id/eip155.rs]: Explained in d44a05581fe064d084da842dcf3715fd8ac855a3

File: src/model/types/account_id/caip10.rs (hunk 1 above)
Comment (chris13524) [reply src/model/types/account_id/caip10.rs]: Explained in d44a05581fe064d084da842dcf3715fd8ac855a3

**Pull request merged**